stub/src/main/java/io/grpc/stub/CallStreamObserver.java

```diff
@@ -19,20 +19,25 @@
 import io.grpc.ExperimentalApi;
 
 /**
- * A refinement of StreamObserver provided by the GRPC runtime to the application that allows for
- * more complex interactions with call behavior.
+ * A refinement of StreamObserver provided by the GRPC runtime to the application (the client or
+ * the server) that allows for more complex interactions with call behavior.
  *
- * <p>In any call there are logically two {@link StreamObserver} implementations:
+ * <p>In any call there are logically three {@link StreamObserver} implementations:
  * <ul>
- *   <li>'inbound' - which the GRPC runtime calls when it receives messages from the
- *   remote peer. This is implemented by the application.
+ *   <li>'inbound' - which the GRPC runtime calls when it receives messages from the server. This
```

Comment:
This 'inbound' `StreamObserver` concept also exists in server-side. As an example, you can check [here](https://github.com/grpc/grpc-java/blob/6b04fc18f30deb3519c389fe457b99cf5a8d99d0/examples/src/main/java/io/grpc/examples/manualflowcontrol/ManualFlowControlServer.java#L38).

Reply:
Thanks @ran-su for pointing to this example. A comment in this example suggests that isReady may turn to false not only when the remote peer (the client) is not able to receive more messages, but also when they have sent us a lot of messages to overflow the `receive buffer`. Should it be reflected in the Javadoc for `CallStreamObserver.isReady()` or `setOnReadyHandler()`? Or was it actually "send buffer", not "receive buffer" which should be mentioned in this comment:
https://github.com/grpc/grpc-java/blob/9e02cf089ee8b3ff635c8531039cfe87da417779/examples/src/main/java/io/grpc/examples/manualflowcontrol/ManualFlowControlServer.java#L92-L98

Reply:
In here, server is only response for manage its own ability to receive messages. If client-side's send buffer fills up there are other system mechanisms to handle it. 

This example is manual flow control example, which may be considered advanced usage of gRPC. IMO, since this is not for all gRPC users, keep all related information in a more detailed example may be better for both user who don't need this feature and for user who actually want this feature.

Reply:
How a series of onNext() -> request(1) -> onNext() -> request(1) calls may lead to overflow of server's receive buffer?

Reply:
In the code I linked, `disableAutoInboundFlowControl()` is called above which disables the ` onNext() -> request(1) -> onNext() -> request(1)` cycle. In the case of `disableAutoInboundFlowControl`, it up to the user of gRPC to call `request()`.

Reply:
In the comment that I excerpted, I think it talks about manual calls already. `serverCallStreamObserver.request(1);`, the very next line after that comment. So yes, user calls `serverCallStreamObserver.request(1)` (strictly once, as ensured in this line:
https://github.com/grpc/grpc-java/blob/9e02cf089ee8b3ff635c8531039cfe87da417779/examples/src/main/java/io/grpc/examples/manualflowcontrol/ManualFlowControlServer.java#L61

Then gRPC runtime calls incoming StreamObserver's `onNext()`, and so on, in a cycle (as far as I understand). How the incoming server's receive buffer may steadily grow and become overflown in this case?

The comment I excerpted above would make more sense to me if it was talking about server's send buffer, e. g. that gRPC runtime makes the server to throttle by means of setting `isReady` to false if the server's send buffer is full, e. g. client doesn't keep up with responses.

Reply:
In the example we are demonstrating a correct way of using those features. If the user is not careful, they may do something equal to `request(some_very_big_number)`, in that case the server receive buffer may get full.

Reply:
@ran-su thanks. I've updated a comment in ManualFlowControlServer.java hopefully to reflect that less ambiguously. I've also removed the usage of `AtomicBoolean` there because it took me quite some time of attempts to understand why a CAS operation was needed there in onReadyHandler before I realized that was just a coding convenience.

I've also updated the Javadoc for `CallStreamObserver`, please review.

Questions 1. and 3. from [the message above](https://github.com/grpc/grpc-java/pull/6561#issue-355749164) also remain.